tests/lib/uicc_lwm2m/src/main.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdarg.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <modem/uicc_lwm2m.h>
+
+#include "unity.h"
+
+struct at_csim_cmd_rsp {
+	char *command;
+	char *response;
+};
+
+static struct at_csim_cmd_rsp *at_csim_command;
+static int at_csim_command_size;
+static int at_csim_command_num;
+
+static struct at_csim_cmd_rsp at_csim_application_not_found[] = {
+	/* Open a logical channel 1 */
+	{ "AT+CSIM=10,\"0070000001\"", "+CSIM: 6,\"019000\"" },
+	/* Select PKCS#15 on channel 1 using the default AID */
+	{ "AT+CSIM=36,\"01A404040CA000000063504B43532D313500\"", "+CSIM: 4,\"6A82\"" },
+	/* Close the logical channel */
+	{ "AT+CSIM=8,\"01708001\"", "+CSIM: 4,\"9000\"" }
+};
+
+static struct at_csim_cmd_rsp at_csim_bootstrap_not_found[] = {
+	/* Open a logical channel 1 */
+	{ "AT+CSIM=10,\"0070000001\"", "+CSIM: 6,\"019000\"" },
+	/* Select PKCS#15 on channel 1 using the default AID */
+	{ "AT+CSIM=36,\"01A404040CA000000063504B43532D313500\"",
+	  "+CSIM: 62,\"621B8202412183025031A5038001718A01058B036F06028002003888009000\"" },
+	/* Select EF(ODF) (path 5031) */
+	{ "AT+CSIM=20,\"01A40804047FFF503100\"",
+	  "+CSIM: 62,\"621B8202412183025031A5038001718A01058B036F06028002003888009000\"" },
+	/* Read EF(ODF) */
+	{ "AT+CSIM=10,\"01B0000080\"", "+CSIM: 42,\"A706300404024407A506300404024404FFFFFF9000\"" },
+	/* Select (path 4407) */
+	{ "AT+CSIM=20,\"01A40804047FFF440700\"",
+	  "+CSIM: 62,\"621B8202412183024407A5038001718A01058B036F0602800200C888009000\"" },
+	/* Read record 4407 (empty) */
+	{ "AT+CSIM=10,\"01B0000080\"", "+CSIM: 42,\"FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF9000\"" },
+	/* Close the logical channel */
+	{ "AT+CSIM=8,\"01708001\"", "+CSIM: 4,\"9000\"" }
+};
+
+static struct at_csim_cmd_rsp at_csim_success[] = {
+	/* Open a logical channel 1 */
+	{ "AT+CSIM=10,\"0070000001\"", "+CSIM: 6,\"019000\"" },
+	/* Select PKCS#15 on channel 1 using the default AID */
+	{ "AT+CSIM=36,\"01A404040CA000000063504B43532D313500\"",
+	  "+CSIM: 82,\"622582027821840CA000000063504B43532D31358A01058B032F060CC60990014083010183010A9000\"" },
```

Comment:
The compliance check for the long lines in this file is entertaining. If I break the line with or without a trailing backslash I get two different warnings. I'm not sure how to fix this without some `#define` hacks or a checkpatch `--exclude`.

```suggestion
	  "+CSIM: 82,\"622582027821840CA000000063504B43532D31358A01058B032F060CC60990014083010183"
	  "010A9000\"" },
```
```
WARNING:MISSING_SPACE: break quoted strings at a space character
#1227: FILE: tests/lib/uicc_lwm2m/src/main.c:60:
+	  "+CSIM: 82,\"622582027821840CA000000063504B43532D31358A01058B032F060CC60990014083010183"
+	  "010A9000\"" },
```

```suggestion
	  "+CSIM: 82,\"622582027821840CA000000063504B43532D31358A01058B032F060CC60990014083010183" \
	  "010A9000\"" },
```
```
WARNING:LINE_CONTINUATIONS: Avoid unnecessary line continuations
#1226: FILE: tests/lib/uicc_lwm2m/src/main.c:59:
+	  "+CSIM: 82,\"622582027821840CA000000063504B43532D31358A01058B032F060CC60990014083010183" \
```

Reply:
I have made checkpatch `--exclude tests/lib/uicc_lwm2m` for now.